src/Stripe.net/Entities/StripeTopup.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Stripe.Infrastructure;
+
+namespace Stripe
+{
+    public class StripeTopup : StripeEntityWithId
+    {
+        [JsonProperty("object")]
+        public string Object { get; set; }
+
+        /// <summary>
+        /// A positive integer in the smallest currency unit (e.g., 100 cents to top up $1.00 or 100 to topup ¥100, a 0-decimal currency) representing how much to top up. The minimum amount is $0.50 US or equivalent in Top-up currency.
+        /// </summary>
+        [JsonProperty("amount")]
+        public int Amount { get; set; }
+
+        #region Expandable Balance Transaction
+        /// <summary>
+        /// ID of the balance transaction that describes the impact of this Top-up on your account balance (not including refunds or disputes).
+        /// </summary>
+        public string BalanceTransactionId { get; set; }
+
+        [JsonIgnore]
+        public StripeBalanceTransaction BalanceTransaction { get; set; }
+
+        [JsonProperty("balance_transaction")]
+        internal object InternalBalanceTransaction
+        {
+            set
+            {
+                StringOrObject<StripeBalanceTransaction>.Map(value, s => BalanceTransactionId = s, o => BalanceTransaction = o);
+            }
+        }
+        #endregion
+
+        [JsonProperty("created")]
+        [JsonConverter(typeof(StripeDateTimeConverter))]
+        public DateTime Created { get; set; }
+
+        [JsonProperty("expected_availability_date")]
+        [JsonConverter(typeof(StripeDateTimeConverter))]
+        public DateTime? ExpectedAvailabilityDate { get; set; }
```

Comment:
Weirdly this value is null in Test so I had to make it optional with the `?`. I feel like we should consider making all properties always optional to be resilient to changes on Stripe's end.

Reply:
I think what you have here is correct given that it's also marked optional in the backend too.

I'm not too sure that we should go around applying this too liberally because the `?` does require a little extra code to use for the user (making it a little less sound ergonomically).

On that note though. I was cross-referencing this with the API resource and noticed these fields are also optional:

* `BalanceTransactionId`
* `Description`
* `FailureCode`
* `FailureMessage`
* `StatementDescriptor`

Should we add `?` for them as well? These are all strings so I think the answer is "no" when comparing to other optional properties throughout the library, but it seems a little strange that nullable is only used for certain types.

Reply:
Yeah I think for strings it's fine. But it'd be great to figure out a better way than throwing the `?` when a test breaks because it means we never fully test the edge-cases